lib/graphql/type_kinds.rb

```diff
@@ -16,22 +16,41 @@ def initialize(name, abstract: false, leaf: false, fields: false, wraps: false,
         @description = description
       end
 
+      # Is this TypeKind abstract?
+      attr_reader :abstract
+      alias_method :abstract?, :abstract
+      remove_method :abstract
```

Comment:
It's also possible to make a helper to automate these, but then IDEs won't know about the presence of the generated methods.

```ruby
module PredicateHelper # This can be mixed into Module, if you're into monkey-patching
  # Given `"foo?"`, this method generates an optimized method that reads from `@foo`.
  def attr_predicate(predicate_name)
    raise ArgumentError unless predicate_name.end_with?("?")
    
    attr_method_name = predicate_name.to_s.delete_suffix("?")
    attr_reader(attr_method_name)
    alias_method(predicate_name, attr_method_name)
    remove_method(attr_method_name)
  end
end
```